Targets transition mega PR (#2)
USGS-R/ds-pipelines-targets-2-course: responses/17-target-objects.md

> ### What to do when you want to specify a non-build-object input to a function? 

Reviewer: ```suggestion
### What to do when you want to specify a non-target input to a command? 
```
I think this revision works? "non-build-object" felt confusing to me - is it the object thing or the non-build thing or both that's important? And some readers might wonder what a build object is. Also, "command" is broader than "function" and both are supported by **targets**

> Wow, we've gotten this far and haven't written a function that accepts anything other than an object target or a file target. I feel so constrained!

Reviewer: 😆

You can add these
```r
p_width <- 12
p_height <- 7
p_units <- "in"
```

to your `_targets.R` file and then call them in the plot command for your targets, such as
```r
tar_target(
  figure_1_png,
  plot_nwis_timeseries(fileout = "3_visualize/out/figure_1.png", site_data_styled,
                       width = p_width, height = p_height, units = p_units),
  format = "file"
)
```

Objects used in the command for `tar_target()` need to be created somewhere before they are called. It is usually a good idea to put configuration info near the top of the makefile. I would suggest adding the code to create these three objects immediately after your `tar_option_set()` call.

---

Another example of when this object (rather than target) pattern comes in handy is when we want to _force_ a target to rebuild. 

```r
library(targets)

list(
  tar_target(
    work_files,
    {
      dummy <- '2021-04-19'
      item_file_download(sb_id = "4f4e4acae4b07f02db67d22b", 
                         dest_dir = "1_fetch/tmp",
                         overwrite_file = TRUE)
    },
    format = "file",
    packages = "sbtools"
  )
)
```

> By adding this `dummy` object to our `command` argument for the `work_files` target, we can modify the dummy contents any time we want to force the update of `work_files`. Making a change to the text in the `dummy` argument allows us to easily record when we last manually refreshed the pipeline's view of what is downloaded from ScienceBase. You may see the use of these `dummy` arguments in spots where there is no other trigger that would cause a rebuild, such as pulling data from a remote webservice or website when targets has no way of knowing that new data are available on the same service URL.

Reviewer: ```suggestion
By adding this `dummy` object to our `command` argument for the `work_files` target, we can modify the dummy contents any time we want to force the update of `work_files`. Updating the `dummy` argument to today's date allows us to simultaneously force the update and record when we last downloaded the data from ScienceBase. You may see the use of these `dummy` arguments in spots where there is no other trigger that would cause a rebuild, such as pulling data from a remote webservice or website when `targets` has no way of knowing that new data are available on the same service URL.
```
